standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/FileUtils.java

```diff
@@ -654,4 +655,11 @@ void findNext() throws IOException {
       nextFile = null;
     }
   }
+
+  public static RemoteIterator<LocatedFileStatus> listAndFilter(FileSystem fileSystem, Path path, boolean recursive,
+                                                                PathFilter filter) throws IOException {
+    return  RemoteIterators.filteringRemoteIterator(
+            fileSystem.listFiles(path, recursive),
+            status -> filter.accept(status.getPath()));
+  }
```

Comment:
Code isn't formatted. Use `eclipse-styles.xml` present in dev-support in hive repo to format the code